src/swap/types.ts

```diff
@@ -59,6 +69,26 @@ export type QuoteWarning = {
   type?: string; // The type of the warning
 };
 
+export type RawTransactionData = {
+  data: string; // The transaction data
+  from: string; // The sender address
+  gas: string; // The gas limit
+  gasPrice: string; // The gas price
+  to: string; // The recipient address
+  value: string; // The value of the transaction
+};
+
+/**
+ * Note: exported as public Type
+ */
+export type Swap = {
+  approveTransaction?: Transaction; // The approval transaction
```

Comment:
What a developer will do with the Approval Transaction? `approve` is a verb, and it feels more something to do that to read.

Reply:
The response will contain an `approveTransaction` (which needs to be broadcast before `transaction`) in order to approve e.g. `Token A` to be used by the contract in a swap between `Token A` to `Token B`. This is only applicable to ERC-20 tokens.

Reply:
https://metaschool.so/articles/what-are-erc20-approve-erc20-allowance-methods/